### PR TITLE
Improve secondaries, put right manifests after install

### PR DESCRIPTION
**Pull request by patriotyk**

### CODE REVIEW

**Comment by cajun-rat** [comment]
@patriotyk This is failing during the CI build, could you take a look please?

**Comment by cajun-rat** [comment]
(Rebased)

File: src/uptane/uptanerepository.cc [hunk 1]
```diff
@@ -90,9 +92,13 @@ std::pair<uint32_t, std::vector<Uptane::Target> > Repository::getTargets() {
         throw MissMatchTarget("director and image");
```
Comment (cajun-rat) [inline src/uptane/uptanerepository.cc]: I'm struggling to understand why it is necessary to split the targets between the primary and secondary(s) here?

The split is only based on ecu_identifier, so would it make more sense to split them into two groups in SotaUptaneClient::runForever instead?

Comment (patriotyk) [inline src/uptane/uptanerepository.cc]: We cannot download target file from director target, only from image.

**Comment by patriotyk** [comment]
I have updated

**Comment by OYTIS** [comment]
Tested on Acer OTA stick.

**Pull request merged**